torch/nn/modules/lazy.py

```diff
@@ -110,11 +114,12 @@ class LazyModuleMixin:
 
     A final caveat when using lazy modules is that the order of initialization of a network's
     parameters may change, since the lazy modules are always initialized after other modules.
-    This can cause the parameters of a network using lazy modules to be initialized differently
-    than the parameters of a network without lazy modules.
     For example, if the LazyMLP class defined above had a :class:`torch.nn.LazyLinear` module
     first and then a regular :class:`torch.nn.Linear` second, the second module would be
     initialized on construction and the first module would be initialized during the first dry run.
+    This can cause the parameters of a network using lazy modules to be initialized differently
+    than the parameters of a network without lazy modules due to random number generation. Check
```

Comment:
Let's make it more clear by saying like: `as the order of parameter initializations, which often depends on random number generator which is stateful, is different`